docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -82,8 +82,54 @@ After this, the `SearchSourceBuilder` only needs to be added to the
 include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-setter]
 --------------------------------------------------
 
-===== Specifying Sorting
+===== Building queries
+
+Search queries are created using `QueryBuilder` objects. A `QueryBuilder` exists
+ for every search query type supported by Elasticsearch's {ref}/query-dsl.html[Query DSL].
+
+A `QueryBuilder` can be created using its constructor:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-query-builder-ctor]
+--------------------------------------------------
+<1> Create a full text {ref}/query-dsl-match-query.html[Match Query] that matches
+the text "kimchy" over the field "user".
+
+Once created, the `QueryBuilder` object provides methods to configure the options
+of the search query it creates:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-query-builder-options]
+--------------------------------------------------
+<1> Enable fuzzy matching on the match query
+<2> Set the prefix length option on the match query
+<3> Set the max expansion options to control the fuzzy
+    process of the query
+
+`QueryBuilder` objects can also be created using the `QueryBuilders` utility class.
+This class provides helper methods that can be used to create `QueryBuilder` objects
+ using a fluent programming style:
```

Comment:
I am not sure this fluent thing will stay forever, maybe don't mention it? I am even wondering why we keep QueryBuilders. Isn't it the same as Requests, which we said it's not that useful?